subprojects/platform-play/src/main/java/org/gradle/play/internal/twirl/TwirlCompilerFactory.java

```diff
@@ -28,25 +28,40 @@ public static TwirlCompiler create(PlayPlatform playPlatform) {
     public static VersionedTwirlCompilerAdapter createAdapter(PlayPlatform playPlatform) {
         String playVersion = playPlatform.getPlayVersion();
         String scalaCompatibilityVersion = playPlatform.getScalaPlatform().getScalaCompatibilityVersion();
+        VersionedPlayTwirlAdapter playTwirlAdapter = createPlayTwirlAdapter(playPlatform);
         switch (PlayMajorVersion.forPlatform(playPlatform)) {
             case PLAY_2_2_X:
-                return new TwirlCompilerAdapterV22X("2.2.6", scalaCompatibilityVersion);
+                return new TwirlCompilerAdapterV22X("2.2.6", scalaCompatibilityVersion, playTwirlAdapter);
```

Comment:
"2.2.6" here is a Play version, not Twirl version.

Reply:
By the way, that is why I was suggesting to have constructors with default **Twirl** versions.

Reply:
Twirl was not a separate library in Play 2.2.6

Reply:
I've sent https://github.com/gradle/gradle/pull/3353 as a separate PR to remove support for Play 2.2, which will remove any potential for confusion moving forward